src/gcsfilesystem.cpp

```diff
@@ -152,7 +152,6 @@ StatusCode GCSFileSystem::isDirectory(const std::string& path,
         SPDLOG_LOGGER_DEBUG(gcs_logger, "GCS list objects exception {}", ex.what());
     }
 
-    SPDLOG_LOGGER_ERROR(gcs_logger, "Invalid or missing GCS credentials, or directory does not exist - {}", path);
```

Comment:
I think we should move this to line 152 inside the catch clause.

Reply:
We should split this message into two, but I assume this does not exist here for a reason - probably not feedback from GCS. Just removing this message does not seem to be the right call, as there are positive exits from this function.

In what scenarios this message appears and should not? And what is the order of events there?